src/redux/actions/applicationActions.ts

```diff
@@ -50,6 +53,39 @@ export function saveAppSettings(appSettings: IAppSettings): (dispath: Dispatch)
     };
 }
 
+/**
+ * Ensures that a valid security token is associated with the project, otherwise creates one
+ * @param project The project to validate
+ */
+export function ensureSecurityToken(project: IProject):
+    (dispatch: Dispatch, getState: () => IApplicationState) => Promise<IAppSettings> {
+    return async (dispatch: Dispatch, getState: () => IApplicationState) => {
+        const appState = getState();
+        let securityToken = appState.appSettings.securityTokens
+            .find((st) => st.name === project.securityToken);
+
+        if (securityToken) {
```

Comment:
Since you have already declared your function as `async` you do not need to wrap all your returns in a promise... it will automatically be wrapped by the compiler...